cfpb/design-system
# Use real navigation template in the CMS preview pane

The preview pane for editing the Navigation now uses the real `sidebar.html` template so editors can see real HTML links with the site CSS applied, and it will look the same as the front-end.

This was branched off of @contolini's branch to use the components template in the component page preview pane: https://github.com/cfpb/design-system/tree/component-preview

so this can't be merged until that is merged first

## Additions

- `navigationPreviewTemplate.js` loads `sidebar.html` (same template as front-end) for use in Preview pane of Netlify CMS editor. in addition to `props` data there are also `page.section` and `site.baseurl` properties set to output all the navigation for all the sections and to make the `href` on links functional


## Removals

- deleted obsolete navigation-related custom CMS preview pane widgets in favor of built-in Netlify CMS widgets

## Changes

- `sidebar.html` was updated to work with both the preview pane and the front-end. 
  - some elements only appear in the preview pane and are hidden in the final front-end template (such as the primary nav links, and secondary navigation for sections that are not the current page)
  - link elements are only output if there is a value in the slug field. this is helpful when you are editing because you can see in the preview right away if you forgot a slug and the nav item is not a link
- only use built-in Netlify CMS widgets in the navigation config now (lists and strings)
- on the front-end, `header.html` now has an active class on the link to whatever vertical you are currently in. css just underlines it, this can be made pretty later

## Testing

1. browse pages at https://deploy-preview-107--cfpb-design-system.netlify.com/design-system/ and confirm the navigation links are as expected
1. go to https://deploy-preview-107--cfpb-design-system.netlify.com/design-system/admin/#/collections/navigation/entries/navigation and edit navigation fields on left and confirm the preview pane on the right reflects your changes as you make them. try opening the navigation preview pane links in new tabs to confirm they function as expected. **DO NOT SAVE**

## Screenshots


![Screen Shot 2019-07-19 at 3 44 30 PM](https://user-images.githubusercontent.com/702526/61564615-d2a46d00-aa44-11e9-822b-2e4780dde7b4.png)


## Notes

-

## Todos

-

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
